### ISSUE TITLE
Making forecasts with dynamic features

### ISSUE THREAD

**Issue by hidelkys** [Time-MoE/Time-MoE#6]
I have successfully used run_eval.py on a few of the long sequence forecasting data set, and the results look promising. From previous posts, I can see how a univariate sequence can be applied using the model.generate().

Now I'm wondering how we could make forecasts on a multivariate customized data with specified dynamic features and visualize the results or output a pandas dataframe? Any suggestion is appreciated.

**Comment by Maple728** [comment]
Thank you for your interest in our work! Currently, Time-MoE supports only univariate time-series forecasting and does not accommodate forecasting with dynamic features. However, if you wish to forecast a multivariate time series, we recommend converting it into multiple univariate time series and then using Time-MoE for forecasting.

If you have any questions or need further assistance, please feel free to contact us.

**Comment by hidelkys** [comment]
Thanks for your input. When I tried using the code below, the prediction results seem much worse than using run_eval.py script. I assume it's because the data loader module in evaluation actually used all variates with channel independence, right? How can we integrate this when visualizing the results?

`
path = "./dataset/ETT-small/ETTh1.csv"
df = pd.read_csv(path)
model = AutoModelForCausalLM.from_pretrained(
    "./model/",
    device_map="cpu",
    trust_remote_code=True,
)

seqs = torch.tensor(df["OT"].values).float().unsqueeze(0)
max_val = seqs.max(dim=-1, keepdim=True)[0]
normed_seqs = seqs / max_val

# forecast
output = model.generate(normed_seqs, max_new_tokens=prediction_length)  
normed_predictions = output[:, -prediction_length:]  
predictions = normed_predictions * max_val
`

**Comment by Maple728** [comment]
These benchmark datasets use standard normalization. The mean and standard deviation are calculated for each channel using the training split of the dataset and then applied to the test split. For more details, please refer to the process described here: [GitHub link](https://github.com/Time-MoE/Time-MoE/blob/main/time_moe/datasets/benchmark_dataset.py#L12). You can use this as a guide for your implementation.

If you have any questions, please feel free to contact us.
